ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineNewPayload.java

```diff
@@ -194,7 +195,8 @@ public JsonRpcResponse syncResponse(final JsonRpcRequestContext requestContext)
     }
 
     // TODO: post-merge cleanup
-    if (!mergeCoordinator.latestValidAncestorDescendsFromTerminal(newBlockHeader)) {
+    if (!mergeCoordinator.latestValidAncestorDescendsFromTerminal(newBlockHeader)
```

Comment:
I don't think so. From my understanding, this is just to make sure the transition from pow to pos works fine. It seems like a duplicate check to me post-merge because just above this code, we've checked if the new block connects to the current chain so when the code reaches here, we can be sure the new block must decent from the terminal block.